packages/mui-system/src/cssVars/createCssVarsProvider.js

```diff
@@ -108,14 +108,14 @@ export default function createCssVarsProvider(options) {
     }
 
     const calculatedMode = (() => {
-      if (!mode) {
```

Comment:
Reverse the logic to be easier to understand, it groups a related test with its outcome.